### DISCUSSION TITLE
実行するとレイアウトが変わってしまいます！

### DISCUSSION THREAD

**Discussion by KOTARO1209** [lit-iphone-course/lit-iphone-course#135] [Question-original]
## (1) 実現したいこと
タップアプリの作成。
レイアウトの教科書をもとに設定を行えば解決することはわかりますが、そもそもなぜこの現象が発生してしまうのかお聞きしたいです！
## (2) 実現するために自分が試した内容とその結果
教科書を終わらせました
## (3) 調べた内容や資料
連打ゲームの教科書
## (4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
![image](https://user-images.githubusercontent.com/75009903/217989955-900e77fb-d4c4-4caa-81cd-15665de4743c.png)
## (5) GitHubのリンク
https://github.com/KOTARO1209/Renda

**Comment by appleEmily** [comment]
githubに、最新をコミットしてほしい〜そうでないと、最新版のコード、ストーリーボードになっていないのでっ

[考えられる原因]
オートレイアウトの掛け方的にそうなってるんだとおもます。
ちょっとこれだとどういうふうにオートレイアウトかけてるかわからないから最新版コミットしたら教えて〜
<img width="708" alt="image" src="https://user-images.githubusercontent.com/49332380/217992669-0cc8c24b-6d8f-4fc7-8f23-698823a58431.png">
<img width="706" alt="image" src="https://user-images.githubusercontent.com/49332380/217992686-967b619a-244d-49d6-bd30-a961cf6f7aab.png">

**Reply by KOTARO1209** [reply]
すいません！プッシュして閉じる前にどこかクリックしてたかもです。最新版あげました！

**Reply by appleEmily** [reply]
んんーなんかあげられてないっぽい〜！
commitして、自分でgithubを見てみて、写真のところがinitial commitから最新のコミットメッセージが見られるように変わるはずだよ〜

commitする時に、ちゃんと左下の push to remote　押しているかしら？

<img width="928" alt="image" src="https://user-images.githubusercontent.com/49332380/218017022-06a98a73-f493-4104-a4b6-95dc3b11a895.png">

**Reply by KOTARO1209** [reply]
やってみました！反映されていますでしょうか？

**Comment by appleEmily** [comment]
質問してくれていた回答としては、ただオートレイアウトのかけ方がおかしいから。って感じです〜！

今回だと、こんな感じでAutolayoutかけてあげると良いかなと思います。

基本的には、「上からどんどん規制をつけていく」が基本です。

ですが、今回はパーツ的にtapのUILabelとTapButtonは下からで規制をかけました。真ん中のスペースで調整されるようにしています。
もっと細かくなる時は、上から順番にかけていくのが良いと思い

ボタンは、1:1の比率が変わらないように「Aspect Ratio」の規制をかけています！
![Frame 26](https://user-images.githubusercontent.com/49332380/218305871-211b6eb9-64f8-418c-9bad-fd8fd2cc21c9.png)
ます。
